### PR TITLE
[7-zip] Update version to 21.06

### LINKED ISSUE #21567: [7-zip] update to <21.07>
Library name: **7-zip**

New version number: **21.06**

Other information that may be useful (release notes, etc...)
https://7-zip.org/
https://www.7-zip.org/download.html
https://sourceforge.net/p/sevenzip/discussion/45797/thread/b463f34259/
https://www.7-zip.org/history.txt

This is for the `Port Update`.

`Tool Update Request` can be found in #19568

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #21567 

Update [7-zip] to the latest version 21.06

Note: No feature needs to test.

### CODE REVIEW

File: ports/7zip/vcpkg.json [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
   "homepage": "https://www.7-zip.org",
-  "supports": "!(linux | osx | arm | uwp)"
+  "supports": "!(arm | uwp)",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
```
Comment (Osyotr) [inline ports/7zip/vcpkg.json:14]: This is not used by the port

File: ports/7zip/portfile.cmake [hunk 2]
```diff
@@ -14,19 +13,18 @@ vcpkg_extract_source_archive_ex(
     NO_REMOVE_ONE_LEVEL
 )
```
Comment (Osyotr) [inline ports/7zip/portfile.cmake:14]: Apply patch from issue

File: ports/7zip/vcpkg.json [hunk 3]
```diff
@@ -1,8 +1,18 @@
 {
   "name": "7zip",
-  "version-string": "19.00",
+  "version-string": "21.06",
   "port-version": 3,
   "description": "Library for archiving file with a high compression ratio.",
   "homepage": "https://www.7-zip.org",
-  "supports": "!(linux | osx | arm | uwp)"
+  "supports": "!(arm | uwp)",
```
Comment (Osyotr) [inline ports/7zip/vcpkg.json:7]: I think this new version does support arm.

**Review by Osyotr — COMMENTED** [review]
CMakeLists.txt should also be modified to include new asm code

**Comment by FrankXie05** [comment]
Fix by #23029

**Pull request closed without merging**